enterprise_subsidy/apps/api/v2/tests/test_transaction_views.py

```diff
@@ -0,0 +1,560 @@
+            self.set_up_admin()
+        if role == 'operator':
+            self.set_up_operator()
+
+        url = reverse("api:v2:transaction-admin-list-create", args=[subsidy_uuid])
+
+        response = self.client.get(url)
+
+        assert response.status_code == expected_response_status
+
+        list_response_data = response.json()["results"]
+        response_uuids = [tx["uuid"] for tx in list_response_data]
+        # admins and operators can see the initial transactions
+        # of their subsidies' ledgers.
+        self._prepend_initial_transaction_uuid(subsidy_uuid, expected_response_uuids)
+        self.assertEqual(sorted(response_uuids), sorted(expected_response_uuids))
+
+    @ddt.data('admin', 'operator')
+    def test_admin_list_transactions_happy_path_with_filters(self, role):
+        """
+        Test filtering of Transaction records for an admin or operator.
+        """
+        if role == 'admin':
+            self.set_up_admin()
+        if role == 'operator':
+            self.set_up_operator()
+
+        query_params = {
+            'lms_user_id': STATIC_LMS_USER_ID,
+            'content_key': self.content_key_1,
+            'subsidy_access_policy_uuid': self.subsidy_access_policy_1_uuid,
+        }
+        query_string = urllib.parse.urlencode(query_params)
+        url = reverse("api:v2:transaction-admin-list-create", args=[self.subsidy_1_uuid])
+        url += '?' + query_string
+
+        response = self.client.get(url)
+
+        assert response.status_code == status.HTTP_200_OK
+
+        list_response_data = response.json()["results"]
+        response_uuids = [tx["uuid"] for tx in list_response_data]
+        expected_response_uuids = [
+            self.subsidy_1_transaction_1_uuid,
+        ]
+        self.assertEqual(response_uuids, expected_response_uuids)
+
+    def test_list_no_permission_for_customer_responds_with_403(self):
+        """
+        Tests that an authenticated admin without role-based access
+        to a given customer gets a 403 when requesting access to transactions therein.
+        """
+        self.set_up_admin()
+        url = reverse("api:v2:transaction-admin-list-create", args=[self.subsidy_3.uuid])
+
+        response = self.client.get(url)
+
+        assert response.status_code == status.HTTP_403_FORBIDDEN
+
+    def test_list_with_mixed_wildcard_admin_and_learner_access_gets_200(self):
+        """
+        Test list Transactions permissions as an all-access admin even though they still have an
+        enterprise-scoped learner role.
+        """
+        self.set_up_admin()
+        self.set_jwt_cookie([
+            (SYSTEM_ENTERPRISE_ADMIN_ROLE, ALL_ACCESS_CONTEXT),
+            (SYSTEM_ENTERPRISE_LEARNER_ROLE, self.enterprise_1_uuid),
+        ])
+
+        # The all-access admin role assignment should let the admin
+        # user read the transactions for subsidy_3
+        url = reverse("api:v2:transaction-admin-list-create", args=[self.subsidy_3_uuid])
```

Comment:
I think the correct test here would be whatever makes the subsidy under test have the same enterprise customer as what's paired with the learner role in the JWT.  So, either change the learner context to `self.enterprise_2_uuid`, or change the subsidy under test to `self.subsidy_1_uuid`.